### ISSUE TITLE
Need Latest Compatible Tweakwise for 2.2.6 Magento Commerce

### ISSUE THREAD

**Issue by vieeg** [EmicoEcommerce/Magento2Tweakwise-archived#97]
Issue Brief
--
Hi Team,

We are currently using the Tweakwise for search and we got an issue with category permission (https://github.com/EmicoEcommerce/Magento2Tweakwise/issues/69) which we are trying to implement on our website ZEB (https://www.zeb.be<https://www.zeb.be/>) (Magento 2.2.6 Commerce).

Now, to resolve the issue we are trying to upgrade the TW plug-in. While upgrade we saw that TW plug-in is using the feature (classes) related to Magento 2.3.x version which results in the code compilation error, see below

"[ReflectionException] Class Magento\InventoryApi\Api\GetSourcesAssignedToStockOrderedByPriorityInterface does not exist"

Can you please help up in getting the compatible version (Magento 2.2.6 commerce) for the above category permission issue along with Tweakwise Export, we will be very thankful to you.
Environment
--
* PHP Version: 7.1
* Magento Version: 2.2.6
* Tweakwise Version:2.1.4
* Tweakwise Export Version:1.5.1
* Magento Deploy Mode: production
* Third party modules:

**Comment by edwinljacobs** [comment]
Hi,

I know this is not a definitive fix but is it possible for you to create a module where you override the tweakwise-export di.xml? In tweakwise-export di.xml we find the following:
```
<type name="Emico\TweakwiseExport\Model\Write\Products\CollectionDecorator\StockData">
        <arguments>
            <argument name="metaData" xsi:type="object">Magento\Framework\App\ProductMetadataInterface</argument>
            <argument name="stockMapProviders" xsi:type="array">
                <item name="stockItemMapProvider" xsi:type="object">Emico\TweakwiseExport\Model\Write\Products\CollectionDecorator\StockData\StockItemMapProvider\Proxy</item>
                <item name="sourceItemMapProvider" xsi:type="object">Emico\TweakwiseExport\Model\Write\Products\CollectionDecorator\StockData\SourceItemMapProvider\Proxy</item>
            </argument>
        </arguments>
    </type>
```
If you override this argument (in your freshly created module)
```
<item name="sourceItemMapProvider" xsi:type="object">Emico\TweakwiseExport\Model\Write\Products\CollectionDecorator\StockData\SourceItemMapProvider\Proxy</item>
```
To:
```
<item name="sourceItemMapProvider" xsi:type="object">Emico\TweakwiseExport\Model\Write\Products\CollectionDecorator\StockData\StockItemMapProvider\Proxy</item>
```
It should work. Obviously this is not a definitive fix but at least you can continue the upgrade process.
In the meantime we will look for a more permanent fix. I added a module tar which does exactly this, maybe you can give it a shot?
[zeb-tweakwise-export-compat.tar.gz](https://github.com/EmicoEcommerce/Magento2Tweakwise/files/4025537/zeb-tweakwise-export-compat.tar.gz)

**Comment by vieeg** [comment]
Hi Team,

We have installed the above module as suggested but still we are getting the same error as mentioned above. For quick reference please see the attached screenshot.
![image](https://user-images.githubusercontent.com/59474091/71877080-4f8be600-314e-11ea-8b1a-f01eb13b3efe.png)

**Comment by edwinljacobs** [comment]
Hello,

We added compatibility for 2.2.X in the export module (this has been tested against 2.2.10). To make sure it works as intended we created branch: issue/97-compat-2.2.X in https://github.com/EmicoEcommerce/Magento2TweakwiseExport. Could you check if that solves your issue? If it does I will merge it and release it. Obviously the provided module (which sadly didnt work) can be removed from your code base. 

Can you check if this branch works (i.e. compiles and produces a correct feed) on your installation?

With kind regards.

**Comment by edwinljacobs** [comment]
Could you verify if the proposed fix works?

**Comment by vieeg** [comment]
Hi Team,

Branch (issue/97-compat-2.2.X) which provided works without any issue. We are still performing some tests on UAT but it seems like all the features are working as expected.

Thanks for all your support.

**Comment by edwinljacobs** [comment]
The branch has been merged in tweakwise-export master and has been released under tag 1.5.4

https://github.com/EmicoEcommerce/Magento2TweakwiseExport/releases/tag/v1.5.4

With kind regards
